### DISCUSSION TITLE
Make CACHE_DIR configurable (CI use case)

### DISCUSSION THREAD

**Discussion by gone-for-coding** [ansible/ansible-lint#3160] [Ideas] [answered]
Currently CACHE_DIR does not seem to be configurable but defaults to either XDG_CACHE_HOME or ~/.cache according to https://github.com/ansible/ansible-lint/blob/b9d96dff73cc5aee5e99f25298774eed17a9296b/src/ansiblelint/config.py#L26.

In some CI systems it is beneficial to cache the cache to prevent collections from being installed over and over again. Gitlab for example only allows caching from the project directory so you need to adjust XDG_CACHE_HOME to get the cache somewhere else.

As this is not documented (to my knowledge) this might be subject to a change and my "workaround" using XDG_CACHE_HOME fails.

Suggestion: Introduce a parameter, environment variable, configuration file option to make the path configurable instead of XDG_CACHE_HOME.

**Comment by ssbarnea** [comment, marked as answer]
I hope you will not be totally be disappointed by a "nope" answer. I think you should take this with Gitlab folks as that behavior might break caching of most tools. I am just trying to prevent adding extra complexity.

XDG_CACHE_HOME is a standard-enough option IMHO, could be even set just for command calling the linter.

**Reply by briantist** [reply]
(not original poster)
Anecdotally, I've never heard of `XDG_CACHE_HOME` and don't know what the `XDG` part stands for.

But it wouldn't bother me to use that variable if it's documented.

**Reply by gone-for-coding** [reply]
> XDG_CACHE_HOME is a standard-enough option IMHO, could be even set just for command calling the linter.

Thanks for the answer @ssbarnea .
I'm actually good with XDG_CACHE_HOME as it does it's job perfectly. I was only concerned, as it is nowhere documented, that the behaviour might change in the future.

So if you/the team intends to keep XDG_CACHE_HOME my suggestion would be to add it to the official docs and consider this question resolved :)

**Reply by hille721** [reply]
fine for me to use `XDG_CACHE_HOME` to change the cache dir, but the docs are saying that it is relative to the `project_dir` what seems to be not true (https://ansible-lint.readthedocs.io/usage/#caching)

Just opened an issue for that: https://github.com/ansible/ansible-lint/issues/3371
